File: API/PCPartPicker/Models/ShippingAddress.cs
using Microsoft.EntityFrameworkCore;

namespace PCPartPicker.Models
{
    [Owned]
    public class ShippingAddress
    {
        public string RecipientName { get; set; } = "";
        public string Line1 { get; set; } = "";
        public string? Line2 { get; set; }
        public string City { get; set; } = "";
        public string State { get; set; } = "";
        public string PostalCode { get; set; } = "";
        public string Country { get; set; } = ""; // 2-letter code ("US","HU",...)
        public string? Phone { get; set; }
    }
}

File: UI/pc-part-picker/src/model/ShippingAddress.ts
// shipping-address.model.ts
export interface ShippingAddressDto {
  recipientName: string;
  line1: string;
  line2?: string;
  city: string;
  state: string;
  postalCode: string;
  country: string; // e.g., 'HU'
  phone?: string;
  currency: string; // 'eur' if you bill in EUR; 'usd' otherwise
}
